src/editor/mod.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::Path;
+use std::fs::File;
+use std::io::Read;
+
+pub static JS: &'static [u8] = include_bytes!("editor.js");
+pub static ACE_JS: &'static [u8] = include_bytes!("ace.js");
+pub static MODE_RUST_JS: &'static [u8] = include_bytes!("mode-rust.js");
+pub static THEME_DAWN_JS: &'static [u8] = include_bytes!("theme-dawn.js");
+pub static THEME_TOMORROW_NIGHT_JS: &'static [u8] = include_bytes!("theme-tomorrow_night.js");
+
+/// Integration of a JavaScript editor for playpens.
+/// Uses the Ace editor: https://ace.c9.io/.
+/// The Ace editor itself, the mode, and the theme files are the
+/// generated minified no conflict versions.
+///
+/// The `Editor` struct should be used instead of the static variables because
+/// the `new()` method
+/// will look if the user has an editor directory in his source folder and use
+/// the users editor instead
```

Comment:
Hmm, themes in my mind relate to the look and feel of things, while this is new _functionality_. I already feel that the `theme` folder is fairly heavy in how much it has since it has playpen configuration, navigation, etc., in it. If anything, I think playpens should get their own section somewhere in the code and then the editor would go there (I'm just not comfortable doing such large modifications to the existing codebase right now). This right now feels a bit "all JavaScript goes here". I put the editor as a separate item because it's _optional_ and I didn't want it mixed up with everything else too much.

This would also imply that `editor` is a submodule of `theme`. But if you do think this is necessary, let me know.